src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.wasm.xml

```diff
@@ -1,7 +1,14 @@
 <linker>
   <assembly fullname="System.Private.CoreLib">
     <type fullname="System.Numerics.Vector">
-      <method signature="System.Boolean get_IsHardwareAccelerated()" body="stub" value="false" /> 
+      <method signature="System.Boolean get_IsHardwareAccelerated()" body="stub" value="false" />
+    </type>
+    <type fullname="System.Environment">
+      <method signature="System.Boolean get_IsSingleProcessor()" body="stub" value="true" />
```

Comment:
Is this necessary?  Substituting Environment.ProcessorCount to be 1 will make the IsSingleProcessor implementation be `=> 1 == 1;`.

Separately, IsSingleProcessor is internal.  Does the linker issue warnings if such substitutions don't exist?  I'm wondering what happens with this in the future if/when we rename or delete this member.

Reply:
> Substituting Environment.ProcessorCount to be 1 will make the IsSingleProcessor implementation be => 1 == 1

That's not the case (there is no inlining), the implementation will stay unchanged but in this case it's tiny code so probably not worth it.

Reply:
It would be inlined at runtime. And the IL size of the additional call instruction is inconsequential and not worth added complexity.

Reply:
Separately, why don't we replace call sites rather than replacing member bodies?

Reply:
> It would be inlined at runtime.

That's not true. it depends on the codegen.

>why don't we replace call sites in addition to replacing member bodies

It's a different operation with it owns pros&cons

Reply:
> That's not true

If a property that just does `=> 1` doesn't get inlined, we have big problems.

Reply:
> we have big problems

Not really. It depends on the codegen as I wrote before. If an interpreter, AOT or any other codegen decides not to inline because the type has static ctor (or for any other reason) that's their logic. Looking at IL and saying this would be inlined at runtime is therefore incorrect as you don't know which engine and where the code is going to run. I don't think it has any impact on this PR and I'm just stating that CoreCLR != runtime.

Reply:
What "codegen" is an interpreter doing?  And for AOT or any other code generation, I stand by my comment; such trivial properties better be inlined.  C# is designed with the idea that trivial properties get inlined; auto-props demand it (and all of our interactive tooling encourages you to use them), or else very frequently accessed trivial properties on hot paths add non-trivial overheads en mass.

We're in the weeds.

Reply:
> What "codegen" is an interpreter doing? 

ok will switch to executing engine terminology for the future